Core/include/Acts/Seeding/SeedFilter.ipp

```diff
@@ -155,16 +271,81 @@ void SeedFilter<external_spacepoint_t>::filterSeeds_1SpFixed(
   // default filter removes the last seeds if maximum amount exceeded
   // ordering by weight by filterSeeds_2SpFixed means these are the lowest
   // weight seeds
-  for (; it < itBegin + maxSeeds; ++it) {
+  unsigned int numTotalSeeds = 0;
+  for (; it < itBegin + seedsPerSpM.size(); ++it) {
+    // stop if we reach the maximum number of seeds
+    if (numTotalSeeds >= maxSeeds) {
+      break;
+    }
+
     float bestSeedQuality = (*it).first;
 
+    if (m_cfg.seedConfirmation) {
+      // continue if higher-quality seeds were found
+      if (numQualitySeeds > 0 and (*it).second->qualitySeed() == false) {
+        continue;
+      }
+      if (bestSeedQuality < (*it).second->sp[0]->quality() and
+          bestSeedQuality < (*it).second->sp[1]->quality() and
+          bestSeedQuality < (*it).second->sp[2]->quality()) {
+        continue;
+      }
+    }
+
+    // set quality of seed components
     (*it).second->sp[0]->setQuality(bestSeedQuality);
     (*it).second->sp[1]->setQuality(bestSeedQuality);
     (*it).second->sp[2]->setQuality(bestSeedQuality);
 
     outIt = Seed<external_spacepoint_t>{
         (*it).second->sp[0]->sp(), (*it).second->sp[1]->sp(),
         (*it).second->sp[2]->sp(), (*it).second->z()};
+    numTotalSeeds += 1;
+  }
+}
+
+template <typename external_spacepoint_t>
+void SeedFilter<external_spacepoint_t>::checkReplaceSeeds(
```

Comment:
isn't it more efficient to run the the test for all found seeds only once, after all low quality seeds have been created?

Reply:
The order and combination of the seeds makes a big difference in the final accepted seeds. This was the only way I managed to produce the same seeds as in Athena.